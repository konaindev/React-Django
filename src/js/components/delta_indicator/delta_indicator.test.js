import DeltaIndicator from "./index";
import renderer from "react-test-renderer";

describe("DeltaIndicator", () => {
  it("renders left indicator correctly", () => {
    const props = {
      delta: 0.2,
      direction: 1,
      indicatorPos: "left"
    };
    const tree = renderer.create(<DeltaIndicator {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders right indicator correctly", () => {
    const props = {
      delta: -0.5,
      direction: -1,
      indicatorPos: "right"
    };
    const tree = renderer.create(<DeltaIndicator {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
