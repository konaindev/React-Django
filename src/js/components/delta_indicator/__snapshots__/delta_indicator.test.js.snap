// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeltaIndicator renders left indicator correctly 1`] = `
<div
  className="delta-indicator"
>
  <span
    className="delta-indicator__arrow delta-indicator__trend-up"
  >
    <svg
      aria-hidden="true"
      color="inherit"
      focusable="false"
      height={5}
      role="presentation"
      viewBox="0 0 10 5"
      width={10}
    >
      <path
        d="M5 0l5 5H0z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </span>
   
  <span
    className="delta-indicator__delta"
  >
    20pts
  </span>
</div>
`;

exports[`DeltaIndicator renders right indicator correctly 1`] = `
<div
  className="delta-indicator"
>
  <span
    className="delta-indicator__delta"
  >
    -50pts
  </span>
   
  <span
    className="delta-indicator__arrow delta-indicator__trend-down"
  >
    <svg
      aria-hidden="true"
      color="inherit"
      focusable="false"
      height={5}
      role="presentation"
      viewBox="0 0 10 5"
      width={10}
    >
      <path
        d="M5 5l5-5H0z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </span>
</div>
`;
